### PR TITLE
make RSocketError.Custom.errorCode usable

### LINKED ISSUE #137: make RSocketError.Custom.errorCode usable
The `errorCode` field of `RSocketError` made `internal`, it is not accessible from application code.

## Motivation
An `RSocketError.Custom` has an error code associated with it. Allowed `errorCode` value should be in range [0x00000301-0xFFFFFFFE]. The only sane reason to set the code for custom exception is an ability to inspect it later within application.

1. the code value is not usable within application when `errorCode` is internal;
2. it is not clear what minimal allowed value is until `IAE` is thrown. `ErrorCode.CustomMin` constant -- the lower bond -- is also internal.

## Desired solution
Make both `errorCode` and  `ErrorCode.CustomMin` public.

### PR DESCRIPTION
**Pull request by alexvas**
resolves #137

### CODE REVIEW

File: rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/RSocketError.kt [hunk 1]
```diff
@@ -16,30 +16,55 @@
 
 package io.rsocket.kotlin
 
-sealed class RSocketError(internal val errorCode: Int, message: String) : Throwable(message) {
+public sealed class RSocketError(internal val errorCode: Int, message: String) : Throwable(message) {
```
Comment (yschimke) [inline rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/RSocketError.kt:19]: Public is default, why are these all needed?

File: rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/RSocketError.kt [hunk 2]
```diff
@@ -59,35 +84,40 @@ internal fun RSocketError(streamId: Int, errorCode: Int, message: String): Throw
...
-            else                       -> when (errorCode >= ErrorCode.CustomMin || errorCode <= ErrorCode.CustomMax) {
+            else -> when (inCustomRange(errorCode)) {
                 true -> RSocketError.Custom(errorCode, message)
                 false -> IllegalArgumentException("Invalid Error frame in Stream ID $streamId: $errorCode '$message'")
             }
         }
     }
 
-internal object ErrorCode {
+public object ErrorCode {
```
Comment (yschimke) [inline rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/RSocketError.kt:94]: Is this the main part of the fix?

File: rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/RSocketError.kt (hunk 1 above)
Comment (whyoleg) [reply rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/RSocketError.kt:19]: explicit api mode require to specify `public` modifier, to make sure, that all `public` apis are explicit

**Comment by whyoleg** [comment]
The main thing, Im not sure, that we need here to expose a way to create an error with `shifted` custom error code.
IMO, better to just expose `errorCode` on all errors + may be expose minimal allowed custom error code, so some users can create some shifted logic on their side if needed, or use just raw codes

**Comment by alexvas** [comment]
As you see, public `ErrorCode.CustomMax` is also required to allow users to create some shifted logic.

**Review by whyoleg — CHANGES_REQUESTED** [review]
Last remark. WDYT, if we move min/max/range check under `public companion object` of `Custom` error?
Name them like `MinAllowedCode` and `MaxAllowedCode`, and leave `ErrorCode` fully internal.
I think, in this way, we will have a better `public` API, as there will be no strange `ErrorCode` class in public API

**Comment by alexvas** [comment]
Good point. Done.

**Review by whyoleg — APPROVED** [review]
LGTM

**Pull request merged**